tests/testthat/test-printing.R

```diff
@@ -328,8 +330,66 @@ test_that("Inset works for table, ref_footnotes, and main footer", {
   no_ins_v <- sapply(no_inset_part, function(x) substr(x, 1, general_inset), USE.NAMES = FALSE)
   ins_v <- sapply(inset_part, function(x) substr(x, 1, general_inset), USE.NAMES = FALSE)
   result <- lapply(list(no_ins_v, ins_v), function(x) all(lengths(regmatches(x, gregexpr(" ", x))) == general_inset))
-  
+
   expect_false(result[[1]]) # No inset
   expect_true(result[[2]]) # Inset
   expect_true(all(vec_tt[sep_index + 1] == "   =============================="))
 })
+
+test_that("Cell and column label wrapping works in printing", {
+    # Set colwidths vector
+    clw <- c(5, 7, 6, 6) + 12
+
+    # Checking in detail if Cell values did wrap correctly
+    result <- toString(matrix_form(tt_for_wrap[10, 1], TRUE), widths = c(10, 8), col_gap = 2)
+    splitted_res <- strsplit(result, "\n")[[1]]
+
+    # First column (rownames) has widths 10 and there is colgap 2
+    expect_identical(.count_str_for_tests(splitted_res[1], " "), 10L + 2L)
+
+    # First column label is 8 char
+    expect_identical(.count_str_for_tests(splitted_res[1], " ", TRUE), 8L)
+
+    # Separator is at the right place and colnames are wrapped
+    # expect_identical(splitted_res[7], "————————————————————")
+    expected <- c("            Incredib",
+                  "            ly long ",
+                  "             column ",
+                  "            name to ",
+                  "               be   ")
+    # expect_identical(splitted_res[1:5], expected)
+
+    # String replacement of NAs wider than expected works with cell wrapping
+    expected <- c("Mean         A very ",
+                  "              long  ",
+                  "            content ",
+                  "            to_be_wr",
+                  "            apped_an",
+                  "            d_splitt",
+                  "               ed   ")
+    expect_identical(splitted_res[8:14], expected)
+
+    # Testing if footers are not affected by this
+    expect_identical(splitted_res[17], main_footer(tt_for_wrap))
+
+    # Works for row names too
+    result <- toString(matrix_form(tt_for_wrap[6, 1], TRUE), widths = c(10, 8), col_gap = 2)
+    splitted_res2 <- strsplit(result, "\n")[[1]]
+    expected <- c("BLACK OR            ",
+                  "AFRICAN             ",
+                  "AMERICAN            ")
+    expect_identical(splitted_res2[8:10], expected)
+
+    # Test if it works with numeric values
+    tt_simple <- basic_table() %>%
+        analyze("AGE", format = "xx.xxxx") %>%
+        build_table(ex_adsl)
+    result <- toString(matrix_form(tt_simple), widths = c(2, 3), col_gap = 1)
+    sre3 <- strsplit(result, "\n")[[1]]
+    expected <- c("   all", "——————", "   obs", "Me 34.", "an 88 ")
```

Comment:
This is testing the buggy behavior. I'll be pushing a fix to that shortly. You'll need to rebuild the test after that goes in so it enforces the right behavior.

Reply:
done